CMakeLists.txt

```
@@ -1617,8 +1617,14 @@ if(HEX_FILES_TO_MERGE)
   message(VERBOSE "Merging hex files: ${HEX_FILES_TO_MERGE}")
 endif()
 
-if(EMU_PLATFORM)
-  include(${ZEPHYR_BASE}/cmake/emu/${EMU_PLATFORM}.cmake)
+if(SUPPORTED_EMU_PLATFORMS)
+  foreach(EMU_PLATFORM ${SUPPORTED_EMU_PLATFORMS})
+    if(EXISTS "${ZEPHYR_BASE}/cmake/emu/${EMU_PLATFORM}.cmake")
+      include(${ZEPHYR_BASE}/cmake/emu/${EMU_PLATFORM}.cmake)
+    else()
+      message(WARNING "CMake file for ${EMU_PLATFORM} not found!")
+    endif()
```

Why a warning here ?

The `SUPPORTED_EMU_PLATFORMS` are now specified by the boards `board.cmake` files and is not a user input argument.
So I think this should just fail hard if there is not a corresponding emu platform file.

The biggest issue with CMake warnings is that they don't fail CI builds, meaning someone renaming from `nsim` to `nsim-emu` and not updating  all places will still pass CI.

So I would keep existing behavior of a hard error here:
```suggestion
    include(${ZEPHYR_BASE}/cmake/emu/${EMU_PLATFORM}.cmake)
```

will result in:
```
CMake Error at /projects/github/ncs/zephyr/CMakeLists.txt:1622 (include):
  include could not find requested file:

    /projects/github/ncs/zephyr/cmake/emu/bad-emulator.cmake
```
that will also fail CI.

Instead of:
```
CMake Warning at /projects/github/ncs/zephyr/CMakeLists.txt:1625 (message):
  CMake file for bad-emulator not found!
```
that will pass CI.

That way we ensure that a supported emulator always have a corresponding cmake file.

---

Done